addon/components/mediasuite-check-boxes.js

```diff
@@ -1,29 +1,18 @@
 import Ember from 'ember'
-const { computed } = Ember
+const { computed, get, set } = Ember
 
 export default Ember.Component.extend({
   tagName: 'fieldset',
-  classNames: ['check-boxes'],
-  isChecked: function (option) {
-    let initialState = Ember.A(this.get('initialState'))
-    return (initialState.indexOf(option.value) !== -1)
+  classNames: ['check-boxes', 'group-element-wrap'],
+  selected: computed.filterBy('options', 'checked', true),
+  didReceiveAttrs () {
+    this._super(...arguments)
+    set(this, 'clonedOptions', JSON.parse(JSON.stringify(get(this, 'options'))))
   },
-  // This is supposed to update whenever the checked status changes on each
-  // item, but it is currently broken @see https://github.com/emberjs/ember.js/issues/12475
-  // Workaround: set options to a new reference (e.g. clone array)
-  checkboxes: computed('options', 'options@each.checked', function () {
-    return this.get('options').map(option => {
-      return Ember.ObjectProxy.create({
-        content: option,
-        checked: option.checked ? true : this.isChecked(option)
-      })
-    })
-  }),
-  selected: computed.filterBy('checkboxes', 'checked', true),
-  selectedValues: computed.mapBy('selected', 'content.value'),
   actions: {
-    update () {
-      this.get('update')(this.get('selectedValues'))
+    onUpdate (option, isChecked) {
+      set(option, 'checked', isChecked)
+      get(this, 'update')(get(this, 'clonedOptions'))
```

Comment:
So this is a bit of a stylistic thing, but I think it's kind of important. One of the benefits of the whole DDAU thing, one way bindings, responding to user input directly instead of passively observing data changes, etc... is that you know exactly what happened.

Right here, you're just passing the updated cloned actions. If an action up the component stack needs to action based on which option was just checked, then it would have to do a lot of work/state to figure out what actually changed. This is a major gripe I have with some third-party components.

I think this is important to think about, because this is a shared component, and we can't make assumptions about how this is going to be used.

I think the following is a more flexible way to pass up this action, because users of this component will always get all of the information to fully describe exactly what just happened.

```javascript
get(this, 'update')(get(this, 'clonedOptions'), option, isChecked)
```

Instead of "here's the new state of everything", you get "the user clicked this option, changing its value to this, and here's the resulting state of everything after that action".

IMO 9/10 you're limiting the applicability of this component by not including the action.